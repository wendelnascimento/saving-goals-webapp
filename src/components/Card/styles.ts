import styled from 'styled-components';

export const CardContainer = styled.div`
  background: ${({ theme }) => theme.white};
  padding: 16px;
  border-radius: 4px;
`;
