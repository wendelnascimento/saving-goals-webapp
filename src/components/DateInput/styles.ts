import styled from 'styled-components';

export const Container = styled.fieldset`
  margin: 0;
  padding: 0;
  border: none;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

export const Legend = styled.legend`
  color: ${({ theme }) => theme.almostBlack};
  font-size: 0.7rem;
  font-weight: 500;
  margin-bottom: 4px;
`;

export const InputBox = styled.div`
  display: flex;
  align-items: stretch;
  justify-content: flex-start;
  width: 100%;
`;

export const Field = styled.div`
  padding: 8px 16px;
  background: ${({ theme }) => theme.white};
  border: 1px solid ${({ theme }) => theme.lightGray};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const Month = styled.span`
  font-size: 0.8rem;
  font-weight: 500;
`;

export const Year = styled.span`
  font-size: 0.7rem;
  font-weight: 400;
`;

export const AddonImage = styled.img`
  width: 6px;
`;

export const AddonLeft = styled.button`
  padding: 8px 16px;
  background-color: ${({ theme }) => theme.lightGray};
  border: 1px solid ${({ theme }) => theme.lightGray};
  border-radius: 4px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right: none;
  transition: background-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  margin: 0;

  &:hover {
    cursor: pointer;
  }

  &:active {
    background-color: ${({ theme }) => theme.superlightGray};
  }

  &:last-child {
    > ${AddonImage} {
      transform: rotateY(180deg);
    }
  }
`;

export const AddonRight = styled(AddonLeft)`
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-left: none;
  border-right: 1px solid ${({ theme }) => theme.lightGray};
`;
