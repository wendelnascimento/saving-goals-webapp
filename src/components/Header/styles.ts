import styled from 'styled-components';

export const Bar = styled.header`
  max-width: 100vw;
  width: 100%;
  background: ${({ theme }) => theme.white};
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid ${({ theme }) => theme.superlightGray};
`;
