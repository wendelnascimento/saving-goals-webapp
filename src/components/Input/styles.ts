import styled from 'styled-components';

export const Container = styled.fieldset`
  margin: 0;
  padding: 0;
  border: none;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

export const Legend = styled.legend`
  color: ${({ theme }) => theme.almostBlack};
  font-size: 0.7rem;
  font-weight: 500;
  margin-bottom: 4px;
`;

export const InputBox = styled.div`
  display: flex;
  align-items: stretch;
  justify-content: flex-start;
`;

export const Field = styled.input`
  padding: 8px;
  background: ${({ theme }) => theme.white};
  border: 1px solid ${({ theme }) => theme.lightGray};
  border-radius: 4px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  font-weight: 700;
  font-size: 1.5rem;
  width: 100%;
`;

export const AddonInput = styled.div`
  padding: 8px 16px;
  background: ${({ theme }) => theme.superlightGray};
  color: ${({ theme }) => theme.darkGray};
  font-size: 1rem;
  border: 1px solid ${({ theme }) => theme.lightGray};
  border-radius: 4px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right: none;
  display: flex;
  align-items: center;
  justify-content: center;
`;
