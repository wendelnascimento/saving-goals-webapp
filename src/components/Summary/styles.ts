import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.lightGray};
`;

export const SummaryRow = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px;
`;

export const SummaryTitle = styled.span`
  display: block;
  font-size: 1rem;
  font-weight: 500;
  color: ${({ theme }) => theme.black};
`;

export const SummaryValue = styled.span`
  display: block;
  font-size: 2rem;
  font-weight: 500;
  color: ${({ theme }) => theme.lightBlue};
`;

export const SummaryRowGray = styled(SummaryRow)`
  background: ${({ theme }) => theme.superlightGray};
`;

export const SummaryDescription = styled.span`
  font-size: 0.7rem;
`;
