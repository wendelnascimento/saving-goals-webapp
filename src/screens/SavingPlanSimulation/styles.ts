import styled from 'styled-components';

export const Wrapper = styled.div`
  margin-top: auto;
  margin-bottom: auto;
  max-width: 450px;
  width: 100%;
`;

export const SavingIcon = styled.img`
  width: 32px;
`;

export const Title = styled.h1`
  font-size: 1.5rem;
  margin-bottom: 0;
  margin-top: 8px;
`;

export const Subtitle = styled.h2`
  font-size: 0.7rem;
  font-weight: 300;
  color: ${({ theme }) => theme.gray};
  margin-top: 0;
`;

export const InputRow = styled.div`
  margin-top: 32px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const InputWrapper = styled.div`
  flex: 0 1 50%;

  &:first-child {
    margin-right: 8px;
  }

  &:last-child {
    margin-left: 8px;
  }
`;

export const SummaryContainer = styled.div`
  margin: 24px 0;
`;

export const SubmitButton = styled.button`
  display: block;
  width: calc(100% - 64px);
  margin: 0 auto;
  padding: 16px;
  border: none;
  background: ${({ theme }) => theme.blue};
  color: ${({ theme }) => theme.white};
  font-weight: 500;
  font-size: 1rem;
  transition: background-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  border-radius: 1000px;

  &:hover {
    background: ${({ theme }) => theme.lightBlue};
    cursor: pointer;
  }

  &:focus,
  &:active {
    background: ${({ theme }) => theme.lightBlue};
  }
`;
