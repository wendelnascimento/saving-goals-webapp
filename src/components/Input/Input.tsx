import React from 'react';

import { Props } from '../../types/input';

import { Container, Legend, InputBox, Field, AddonInput } from './styles';

const Input: React.FC<Props> = props => (
  <Container>
    <Legend>{props.label}</Legend>
    <InputBox>
      <AddonInput>{props.addonBefore}</AddonInput>
      <Field value={props.value} onChange={props.onChange} />
    </InputBox>
  </Container>
);

export default Input;
