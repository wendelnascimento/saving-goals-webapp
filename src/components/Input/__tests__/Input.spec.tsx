import React from 'react';
import { shallow, mount } from 'enzyme';

import Input from '../Input';
import { AddonInput, Legend } from '../styles';

describe('<Input />', () => {
  const props = {
    label: 'Test label',
    onChange: jest.fn(),
    value: '',
    addonBefore: '$',
  };

  it('Should render', () => {
    shallow(
      <Input
        label={props.label}
        onChange={props.onChange}
        value={props.value}
        addonBefore={props.addonBefore}
      />,
    );
  });

  it('Should match stored snapshot', () => {
    const wrapper = mount(
      <Input
        label={props.label}
        onChange={props.onChange}
        value={props.value}
        addonBefore={props.addonBefore}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should have an addon on left', () => {
    const wrapper = mount(
      <Input
        label={props.label}
        onChange={props.onChange}
        value={props.value}
        addonBefore={props.addonBefore}
      />,
    );

    expect(wrapper.find(AddonInput).text()).toEqual('$');
  });

  it('Should have the correct label', () => {
    const wrapper = mount(
      <Input
        label={props.label}
        onChange={props.onChange}
        value={props.value}
        addonBefore={props.addonBefore}
      />,
    );

    expect(wrapper.find(Legend).text()).toEqual('Test label');
  });
});
