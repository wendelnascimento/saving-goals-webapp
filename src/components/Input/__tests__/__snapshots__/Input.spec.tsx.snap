// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should match stored snapshot 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  font-size: 0.7rem;
  font-weight: 500;
  margin-bottom: 4px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  padding: 8px;
  border: 1px solid;
  border-radius: 4px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  font-weight: 700;
  font-size: 1.5rem;
  width: 100%;
  margin: 0;
  -webkit-appearance: none;
}

.c3 {
  padding: 8px 16px;
  font-size: 1rem;
  border: 1px solid;
  border-radius: 4px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<Input
  addonBefore="$"
  label="Test label"
  onChange={[MockFunction]}
  value=""
>
  <styled.fieldset>
    <fieldset
      className="c0"
    >
      <styled.legend>
        <legend
          className="c1"
        >
          Test label
        </legend>
      </styled.legend>
      <styled.div>
        <div
          className="c2"
        >
          <styled.div>
            <div
              className="c3"
            >
              $
            </div>
          </styled.div>
          <styled.input
            onChange={[MockFunction]}
            value=""
          >
            <input
              className="c4"
              onChange={[MockFunction]}
              value=""
            />
          </styled.input>
        </div>
      </styled.div>
    </fieldset>
  </styled.fieldset>
</Input>
`;
